packages/sit-onyx/src/components/OnyxNavBar/modules/OnyxNavItem/OnyxNavItem.vue

```diff
@@ -1,42 +1,192 @@
-<script setup lang="ts">
-import { computed } from "vue";
-import { extractLinkProps } from "../../../../utils/router";
-import OnyxExternalLinkIcon from "../../../OnyxExternalLinkIcon/OnyxExternalLinkIcon.vue";
-import OnyxMenuItem from "../OnyxMenuItem/OnyxMenuItem.vue";
+<script lang="ts" setup>
+// this layout component is only used internally for the nav button component
+// to easily switch between mobile and desktop layout
+import arrowSmallLeft from "@sit-onyx/icons/arrow-small-left.svg?raw";
+import { computed, inject, provide } from "vue";
+import { useLink } from "../../../../composables/useLink";
+import { useMoreListChild } from "../../../../composables/useMoreList";
+import { useVModel, type Nullable } from "../../../../composables/useVModel";
+import { injectI18n } from "../../../../i18n";
+import { mergeVueProps } from "../../../../utils/attrs";
+import OnyxButton from "../../../OnyxButton/OnyxButton.vue";
+import {
+  MOBILE_NAV_BAR_INJECTION_KEY,
+  NAV_BAR_IS_TOP_LEVEL_INJECTION_KEY,
+  NAV_BAR_MORE_LIST_INJECTION_KEY,
+} from "../../types";
+import OnyxFlyoutMenu from "../OnyxFlyoutMenu/OnyxFlyoutMenu.vue";
+import OnyxNavItemFacade from "../OnyxNavItemFacade/OnyxNavItemFacade.vue";
+import OnyxNavSeparator from "../OnyxNavSeparator/OnyxNavSeparator.vue";
 import type { OnyxNavItemProps } from "./types";
 
 const props = withDefaults(defineProps<OnyxNavItemProps>(), {
   active: "auto",
+  open: undefined,
 });
 
-defineSlots<{
+const emit = defineEmits<{
   /**
-   * Content of the nav item.
+   * Emitted when the state of mobile children visibility changes.
+   */
+  "update:open": [value: Nullable<boolean>];
+}>();
+
+const slots = defineSlots<{
+  /**
+   * You can replace the label of the menuitem with your custom content.
    */
   default?(): unknown;
+  /**
+   * OnyxNavItem Children can be nested here.
+   */
+  children?(): unknown;
 }>();
 
-const extractedLinkProps = computed(() => {
-  return props.link ? extractLinkProps(props.link) : undefined;
+const { t } = injectI18n();
+
+/**
+ * Controls the open state.
+ */
+const open = useVModel({
+  props,
+  emit,
+  key: "open",
+  initialValue: false,
 });
+
+const hasChildren = computed(() => !!slots.children);
+
+const { isActive } = useLink();
+const active = computed(() => {
+  if (props.active !== "auto") return props.active;
+  return isActive.value(props.link);
+});
+
+const { componentRef, isVisible } = useMoreListChild(NAV_BAR_MORE_LIST_INJECTION_KEY);
+
+const isMobile = inject(
+  MOBILE_NAV_BAR_INJECTION_KEY,
+  computed(() => false),
+);
+
+const isTopLevel = inject(NAV_BAR_IS_TOP_LEVEL_INJECTION_KEY, true);
+provide(NAV_BAR_IS_TOP_LEVEL_INJECTION_KEY, false);
 </script>
 
 <template>
-  <OnyxMenuItem class="onyx-component onyx-nav-item" :active="props.active" :link="props.link">
-    <slot>
-      <span>{{ props.label }}</span>
-      <OnyxExternalLinkIcon v-bind="extractedLinkProps" />
-    </slot>
-  </OnyxMenuItem>
+  <div
```

Comment:
(optional) we could add some logical whitespaces/newlines here to visually separate grouped code to reduce cognitive complexity